src/client.rs

```diff
@@ -279,44 +279,42 @@ impl Client {
     /// See [11.5 New Window](https://w3c.github.io/webdriver/#dfn-new-window) of the editor's
     /// draft standard.
     #[cfg_attr(docsrs, doc(alias = "New Window"))]
-    pub async fn new_window(
-        &mut self,
-        as_tab: bool,
-    ) -> Result<webdriver::response::NewWindowResponse, error::CmdError> {
+    pub async fn new_window(&mut self, as_tab: bool) -> Result<NewWindowResponse, error::CmdError> {
         let type_hint = if as_tab { "tab" } else { "window" }.to_string();
         let type_hint = Some(type_hint);
-        let params = NewWindowParameters { type_hint };
+        let params = webdriver::command::NewWindowParameters { type_hint };
         match self.issue(WebDriverCommand::NewWindow(params)).await? {
             Json::Object(mut obj) => {
                 let handle = match obj
                     .remove("handle")
-                    .and_then(|x| x.as_str().map(String::from))
+                    .and_then(|x| x.as_str().map(WindowHandle::try_from))
                 {
-                    Some(handle) => handle,
-                    None => return Err(error::CmdError::NotW3C(Json::Object(obj))),
+                    Some(Ok(handle)) => handle,
+                    _ => return Err(error::CmdError::NotW3C(Json::Object(obj))),
                 };
 
-                let typ = match obj
-                    .remove("type")
-                    .and_then(|x| x.as_str().map(String::from))
-                {
-                    Some(typ) => typ,
+                let typ = match obj.get("type").and_then(|x| x.as_str()) {
```

Comment:
Is that the only important field on a new window response?

Reply:
Yup. I haven't changed semantics here, the previous code using `webdriver` types was:
```rust
                let typ = match obj
                    .remove("type")
                    .and_then(|x| x.as_str().map(String::from))
                {
                    Some(typ) => typ,
                    None => return Err(error::CmdError::NotW3C(Json::Object(obj))),
                };

                Ok(webdriver::response::NewWindowResponse { handle, typ })
```

I've only typed the `typ` better [according to the spec](https://www.w3.org/TR/webdriver/#dfn-new-window):
> 7. Let type be "tab" if the newly created window shares an OS-level window with the current browsing context, or "window" otherwise.